sympy/polys/tests/test_polytools.py

```diff
@@ -300,6 +302,34 @@ def test_Poly_from_expr():
     assert Poly.from_expr(y + 5, x, y, domain=ZZ).rep == DMP([[ZZ(1), ZZ(5)]], ZZ)
 
 
+def test_Poly_rootof_extension():
+    r1 = rootof(x**3 + x + 3, 0)
+    r2 = rootof(x**3 + x + 3, 1)
+    K1 = QQ.algebraic_field(r1)
+    K2 = QQ.algebraic_field(r2)
+    assert Poly(r1, y) == Poly(r1, y, domain=EX)
+    assert Poly(r2, y) == Poly(r2, y, domain=EX)
+    assert Poly(r1, y, extension=True) == Poly(r1, y, domain=K1)
+    assert Poly(r2, y, extension=True) == Poly(r2, y, domain=K2)
+
+
+@tooslow
+def test_Poly_rootof_extension_primitive_element():
+    r1 = rootof(x**3 + x + 3, 0)
+    r2 = rootof(x**3 + x + 3, 1)
+    K12 = QQ.algebraic_field(r1 + r2)
+    assert Poly(r1*y + r2, y, extension=True) == Poly(r1*y + r2, y, domain=K12)
+
+
+@XFAIL
+def test_Poly_rootof_same_symbol():
+    # XXX: This fails because r1 contains x.
+    r1 = rootof(x**3 + x + 3, 0)
```

Comment:
This case is discussed in more detail in gh-26808. Better to discuss there.

Here I added a test and then made it XFAIL because it failed and the fix would be more involved than the basic issue that this PR is addressed at.

Please don't allow the very obvious fix that this PR provides to be sidetracked by other issues that should be fixed elsewhere. I can remove this test if needed but the simple fix here should be merged regardless of other issues.